Swiftification/URL.swift

```diff
@@ -25,7 +25,7 @@ import Foundation
 public extension URL {
     
     func replacing(queryParameters parameters: [String: String]) -> URL? {
-        guard !parameters.isEmpty, var components = URLComponents(url: self, resolvingAgainstBaseURL: false) else { return nil }
+        guard !parameters.isEmpty, var components = URLComponents(url: self, resolvingAgainstBaseURL: false) else { return self }
```

Comment:
This could fail and return `self` for two different reasons: no parameters (should just return `self`), or can't parse (should return `nil`?)... I'd either split this into two guards, or not return an optional and always fall back to `self`.

Reply:
Good call.